atecc608a_se.c

```diff
@@ -0,0 +1,224 @@
+#include "atecc608a_se.h"
+
+static ATCAIfaceCfg atca_iface_config = {
+    .iface_type = ATCA_I2C_IFACE,
+    .devtype = ATECC608A,
+    .atcai2c.slave_address = 0xC0,
+    .atcai2c.bus = 2,
+    .atcai2c.baud = 400000,
+    .wake_delay = 1500,
+    .rx_retries = 20,
+};
+
+int hash_sha256(const uint8_t *input, size_t input_size,
+                        const uint8_t *expected_hash, size_t expected_hash_size)
+{
+    uint8_t actual_hash[ATCA_SHA_DIGEST_SIZE] = {0};
+
+    printf("**********************************************\n");
+    printf("SHA-256:\n\n");
+    atcab_printbin_label("Input: ", (uint8_t *)input, input_size);
+    atcab_printbin_label("Expected Hash: ", (uint8_t *)expected_hash, expected_hash_size);
+    ASSERT_STATUS(atcab_init(&atca_iface_config), ATCA_SUCCESS);
+    ASSERT_STATUS(atcab_hw_sha2_256(input, input_size, actual_hash), ATCA_SUCCESS);
+    ASSERT_STATUS(atcab_release(), ATCA_SUCCESS);
+    atcab_printbin_label("Actual Hash: ", actual_hash, ATCA_SHA_DIGEST_SIZE);
+    ASSERT_STATUS(memcmp(actual_hash, expected_hash, sizeof(actual_hash)), 0);
+    printf("Success!\n\n");
+
+    return 0;
+}
+
+int read_serial_number(void)
+{
+    uint8_t serial[ATCA_SERIAL_NUM_SIZE];
+
+    printf("**********************************************\n");
+    ASSERT_STATUS(atcab_init(&atca_iface_config), ATCA_SUCCESS);
```

Comment:
The driver code should not use `ASSERT_STATUS`, which prints a debugging message and returns `-1`. It should return an appropriate error code when an error happens, e.g. `PSA_ERROR_HARDWARE_FAILURE` if the hardware returns an unexpected error code. The driver may contain debug prints, but these should be controlled by an `#ifdef`.